skglm/tests/test_estimators.py

```diff
@@ -531,6 +534,95 @@ def test_warm_start(estimator_name):
     model.fit(X, y)  # already fitted + warm_start so 0 iter done
     np.testing.assert_equal(0, model.n_iter_)
 
+@pytest.mark.parametrize("fit_intercept, issparse",
+                         product([False, True], [False, True]))
+def test_GroupLasso_estimator(fit_intercept, issparse):
+
+    grp_indices, grp_ptr = grp_converter(groups, X.shape[1]) 
+    n_groups = len(grp_ptr)-1
+    weights = np.ones(n_groups)
+    alpha = _alpha_max_group_lasso(X, y, grp_indices, grp_ptr, weights)/10.
+    estimator_ours = GroupLasso(groups=groups, alpha=alpha, tol=tol, 
+                                weights=weights, fit_intercept=fit_intercept,
+                                positive=False)
+    estimator_celer = GroupLasso_celer(groups=groups, alpha=alpha, tol=tol, 
+                                       weights=weights, fit_intercept=fit_intercept)
+
+    X_ = csc_matrix(X) if issparse else X
+
+    estimator_celer.fit(X_, y)
+    estimator_ours.fit(X_, y)
+    coef_celer = estimator_celer.coef_
+    coef_ours = estimator_ours.coef_
+
+    np.testing.assert_allclose(coef_ours, coef_celer, atol=1e-6)
+    np.testing.assert_allclose(
+        estimator_celer.intercept_, estimator_ours.intercept_, rtol=1e-4)
+    if fit_intercept:
+        np.testing.assert_array_less(1e-4, estimator_ours.intercept_)
+
+
+@pytest.mark.parametrize("fit_intercept, issparse",
+                         product([False, True], [True, False]))
+def test_GroupLasso_estimator_positive(fit_intercept, issparse):
+
+    grp_indices, grp_ptr = grp_converter(groups, X.shape[1]) 
+    n_groups = len(grp_ptr)-1
+    weights = np.ones(n_groups)
+    alpha = _alpha_max_group_lasso(X, y, grp_indices, grp_ptr, weights)/10.
+    estimator_ours = GroupLasso(groups=groups, alpha=alpha, tol=tol, 
+                                weights=weights, fit_intercept=fit_intercept,
+                                positive=True)
+
+    X_ = csc_matrix(X) if issparse else X
+
+    estimator_ours.fit(X_, y)
+
+    assert np.all(estimator_ours.coef_>=0)
```

Comment:
we try to use np.testing functions (more explicit error messages), here np.testing.assert_array_less will do the job